pygsti/circuits/circuit.py

```diff
@@ -3405,6 +3405,30 @@ def two_q_gate_count(self):
         """
         return self.num_nq_gates(2)
 
+    @property
+    def num_gates(self):
+        """
+        The number of gates in the circuit.
+
+        Returns
+        -------
+        int
+        """
+        if self._static:
+            def cnt(lbl):  # obj a Label, perhaps compound
+                if lbl.is_simple():  # a simple label
+                    return 1 if (lbl.sslbls is not None) else 0
+                else:
+                    return sum([cnt(sublbl) for sublbl in lbl.components])
+        else:
+            def cnt(obj):  # obj is either a simple label or a list
+                if isinstance(obj, _Label):  # all Labels are simple labels
+                    return 1 if (obj.sslbls is not None) else 0
+                else:
+                    return sum([cnt(sub) for sub in obj])
+
+        return sum([cnt(layer_lbl) for layer_lbl in self._labels])
+
```

Comment:
If I understand this correctly, this is computing the number of explicitly included (i.e., not implicit idle) gates in the circuit? I.e., this is different to self.depth * self.width right? If so, (1) this seems like a useful function to have, and (2) it's debatable whether this is what should be computed by the RB code. I suspect that what should be returned is this *and* the circuit size (width * depth)? Some users might also be interested in the 2-qubit gate count, so that could also be computed and all wrapped up in a dictionary?

Reply:
You are correct, this is explicitly included gates and not depth*width (which already exists as `circuit.size`). FYI, the 2-qubit gate count also exists in that you can do `circuit.num_nq_gates(2)`.

Computing all three and wrapping it in a dict is a pretty easy and self-documenting solution, so I'll implement that. :)

Reply:
This is now what the output of the `average_native_gates_per_clifford()` looks like:

<img width="396" alt="Screenshot 2024-05-30 at 1 04 16 PM" src="https://github.com/sandialabs/pyGSTi/assets/72409998/08589477-22a4-49da-9fde-3f75a390810e">